### DISCUSSION TITLE
really need: please support edit the music tag

### DISCUSSION THREAD

**Discussion by zjyg1993** [navidrome/navidrome#3103] [Ideas]
as the title wrote,i downloaded some music from web ,but sometimes , the music tag is wrong ,and navidrome show the wrong title or other tag ,can you suppot edit the tag  in the playlist interface ? it's very useful for me , even more people will like the function. thank you

**Comment by NeuroDawg** [comment]
probably not going to happen when there are so many tag editors already out there. just get mp3tag and fix the tags yourself.

**Comment by deluan** [comment]
Sorry, but there are no plans to implement tag editing in Navidrome, see https://www.navidrome.org/docs/faq/#-how-can-i-edit-my-music-metadata-id3-tags-how-can-i-renamemove-my-files

As @NeuroDawg, there are plenty of excellent taggers out there, for all platforms.

**Comment by sobchenyuk** [comment]
**Tag editing is now available!** 🎉

I've implemented a comprehensive tag editing solution by extending Navidrome with a new tech stack:

**🔗 Repository:** https://github.com/sobchenyuk/navidrome

**✨ Current Features:**
- Full tag editing through web interface
- Batch operations for multiple files
- Real-time preview of changes
- Support for all common audio formats

**🚀 Planned Features:**
- Encoding fix for Cyrillic tags (CP1251 ↔ UTF-8)
- Advanced batch processing with workers
- Enhanced UI with progress tracking

**🛠 Tech Stack:**
- **Backend:** NestJS + TypeScript + GraphQL
- **Frontend:** React + Material-UI + Apollo Client
- **Database:** SQLite3 (existing Navidrome database)
- **Audio Processing:** music-metadata + node-id3 + flac-metadata2
- **Build Tools:** Vite + TypeScript
- **Architecture:** GraphQL API with real-time capabilities

**🎯 Technical approach:**
- Extended Navidrome core with additional service layer
- Modern React UI components with Material Design
- RESTful API for tag operations
- Non-destructive workflow with backup capabilities

This maintains Navidrome's philosophy while adding the tag editing functionality many users have requested. The extension works seamlessly with existing Navidrome installations.

Feel free to try it out and provide feedback!

**Comment by farfromrefug** [comment]
@deluan @sobchenyuk will this ever make it s way in main? that fork has not been updated in 5 months. and it would be really sad to such a wanted feature not make it in as it is now implemented. Though obviously it would some changes (more than 5 months old already)

**Comment by deluan** [comment]
@farfromrefug No, there are no plans to merge this, and there are no plans to add Tag editing capabilities to Navidrome, as I mentioned above:

> Sorry, but there are no plans to implement tag editing in Navidrome, see https://www.navidrome.org/docs/faq/#-how-can-i-edit-my-music-metadata-id3-tags-how-can-i-renamemove-my-files
> 
> As @NeuroDawg, there are plenty of excellent taggers out there, for all platforms.

Feel free to use @sobchenyuk's fork, but at this point it is just that: a fork. Not something I'm evaluating to merge back into Navidrome.

**Reply by farfromrefug** [reply]
@deluan i dont want to use or maintain a different fork. But anyway i didnt see it was actually using full different tech stack, which makes it almost impossible to merge anyway.
I do understand your point about using other softwares.  But as a software dev myself there is always the leveraging of having 10 different tools (with maintenance cost, memory cost...) to handle related tasks, or finding a way to merge (even though i dont like that word here) them into one solution.
For example i dont know how feasible it would be, put plugin tags edition on an installed beet for example would be a nice way to have a GUI to quickly edit tags from navidrome while not having navidrome reimplement the whole thing. 
Just throwing that here even though i dont think it would change your mind.
Thanks for taking the time to answer me

**Comment by yashipashi** [comment]
I am not sure why there is such intransigence towards this feature? There has to be a separation between the navidrome backend which has a very specific goal and the web UI, which in principle you'd be happy to do EVERYTHING from: any possible manipulation of your music library. This is especially so if you are working remotely and do not wish to bother SSHing into your serve just to edit a tag. This is ESPECIALLY so if you have a 4 tera byte library spanned across multiple hard drives after backup disasters where no file has any meaningful name and you randomly come across a recording while shuffling through your playlist and it is INVALUABLY useful to edit WITHIN the web interface.

I urge you to reconsider. I asked ChatGPT to write for me a tag editor patch for navidrome and within 30 minutes I had a working binary of Navidrome running and able to do this. I don't see why it would be such a travesty if this were actually audited and maintained by the community, and available to everyone at the flip of a switch.

**Comment by aqua47** [comment]
Hi, For those who are interested in editing MP3 tags directly in Navidrome, I'm currently working on it.

It's a fork that adds tag editing (titles, albums, artists...). It will remain a fork due to the philosophy of the main project, and it uses the exact same tech stack.

You can check it out here: https://github.com/aqua47/navidrome

**Reply by yashipashi** [reply]
Thank you for doing that! I also did a fork: https://github.com/yashipashi/navidrome with similar feature. The next thing I wanted to try is to deal with duplicat esources to tracks.

**Reply by farfromrefug** [reply]
@aqua47 very nice indeed. The first thing I was looking at with tag edition is like a very quick one op to mark albums as compilation. I feel like it is the biggest pain to see all tracks separated

**Comment by sobchenyuk** [comment]
@deluan @farfromrefug I haven't been developing anything for 9 months—I was drafted into the army, and now I'm just stoking the stove there.
